queries/reflect.go

```diff
@@ -301,24 +301,27 @@ func PtrsFromMapping(val reflect.Value, mapping []uint64) []interface{} {
 func ValuesFromMapping(val reflect.Value, mapping []uint64) []interface{} {
 	ptrs := make([]interface{}, len(mapping))
 	for i, m := range mapping {
-		ptrs[i] = ptrFromMapping(val, m, false).Interface()
+		ptrs[i] = ptrFromMapping(val, m, false)
 	}
 	return ptrs
 }
 
 // ptrFromMapping expects to be passed an addressable struct that it's looking
 // for things on.
-func ptrFromMapping(val reflect.Value, mapping uint64, addressOf bool) reflect.Value {
+func ptrFromMapping(val reflect.Value, mapping uint64, addressOf bool) interface{} {
+	if mapping == 0 {
+		var ignored interface{}
+		return &ignored
```

Comment:
On a second look at this, I want to understand the reason for changing all the type signatures etc.

Couldn't we have just done:
```go
var ignored interface{}
return reflect.ValueOf(&ignored)
```

and had the same effect?

Reply:
Thanks! I'm a rookie on Golang. I can use reflect directly. I was thinking the function of ptrFromMapping will return the type of *interface {}, so we don't need to reflect it again. As my understanding, there no other functions use ptrFromMapping's variables and related methods.

Reply:
And I was thinking if we don't need to use reflect necessarily, we could avoid the performance additional loss.